### PR TITLE
docker: don't build containerd and runc, depend on pkgs that provide it

### LINKED ISSUE #4967: docker>=18.09.0 conflicts with runc and containerd
As of the Docker 18.09.0 update, the docker package now conflicts with the runc and containerd packages. docker-18.06.0 does not conflict. This is due to Docker including binaries that are provided by both runc and containerd. The following is a list of conflicting binaries (and the packages they're in), going by xlocate:

```
# xbps-query -f docker | awk '{print $1}' | xargs -I{} xlocate '^{}$' | sort -k2 | uniq -Df1
containerd-1.1.4_1  /usr/bin/containerd
docker-18.09.0_1    /usr/bin/containerd
containerd-1.1.4_1  /usr/bin/containerd-shim
docker-18.09.0_1    /usr/bin/containerd-shim
containerd-1.1.4_1  /usr/bin/ctr
docker-18.09.0_1    /usr/bin/ctr
docker-18.09.0_1    /usr/bin/runc
runc-1.0.0_6        /usr/bin/runc
```

This seems like a case where the docker package should depend on containerd and runc to provide these binaries, but I'm not sure yet if these are the same between packages or if Docker is just claiming the names (I'm also not sure what version of these Docker uses). Previously, the conflicting binaries all had the docker- prefix, so there was no conflict.

### PR DESCRIPTION
**Pull request by maxice8**
closes #4967

@nilium @CameronNemo

### CODE REVIEW

File: srcpkgs/docker/template [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 pkgname=docker
 version=18.09.3
-revision=1
+revision=2
 wrksrc="${pkgname}-ce-${version}"
 hostmakedepends="git go pkg-config curl cmake"
-makedepends="libbtrfs-devel sqlite-devel device-mapper-devel libseccomp-devel libapparmor-devel libltdl-devel"
+makedepends="libbtrfs-devel sqlite-devel device-mapper-devel libseccomp-devel
+ libapparmor-devel libltdl-devel"
+depends="containerd runc"
```
Comment (nilium) [inline srcpkgs/docker/template]: There's a second depends variable on line 18 -- either that should be updated or its packages should be moved into this one.

Comment (CameronNemo) [reply srcpkgs/docker/template]: relatedly, why does it require git at runtime??

**Comment by nilium** [comment]
Confirmed docker still works with these changes, so aside from the depends thing, I think this looks good.

File: srcpkgs/docker/template (hunk 1 above)
Comment (nilium) [reply srcpkgs/docker/template]: `docker build <path|url>` takes either a path to a context or a URL to a git repository. Wondered the same a while ago -- I don't know how many people use that feature, but it's there.

**Comment by maxice8** [comment]
@thypon This also installs fish completion, and vim Dockerfile syntax

**Comment by thypon** [comment]
@maxice8 seems to be an amazing change!

I still wonder if it won't destabilize docker with incompatible APIs update of runc/containerd, they both have not been known for API stability in the past.

**Comment by maxice8** [comment]
Yes there is the possibility of a runc/containerd update screwing up docker.

for runc we can make use of its testsuite that uses docker.

**Comment by maxice8** [comment]
But yeah in the end there isn't much that can done short of having people updating containerd/runc test against docker.

**Pull request merged**